chore: Adjust Links and documentation

FrankEssenberger opened the pull request:
<!-- Please provide a description of what your change does and why it is needed. -->

Relates SAP/cloud-sdk-backlog#1022.

The API docs were outdated. This PR updates this and aligns a few things:
- Remove the fluff text in the beginning of the API doc (DOCUMENTATION.md) - this just gets outdated and a links section with the title is self explanatory
- Use the same link section in root README.md, package README.md and the DOCUMENTATION.md
- The line spacing in the links sections was to big -> adjusted using a double <br>
- On the landing page we call it SDK for JavaScript (and TypeScript) align this to always use the bracket version since it is subtle

fwilhe commented:
none of the tasks in the BLI are checked off

fwilhe commented on DOCUMENTATION.md:
```diff
@@ -1,16 +1,19 @@
 ## Version: 2.11.0
 
-The SAP Cloud SDK for JavaScript and TypeScript consists of multiple libraries to enable you to write side-by-side extensions on SAP Business Technology Platform.
```
technically our product name is "SAP Cloud SDK for JavaScript"

do we need to disambiguate that it can also be used with typescript? does it make sense to do that in another sentence?

> The SAP Cloud SDK for JavaScript consists of multiple libraries to enable you to write side-by-side extensions on SAP Business Technology Platform.
> It works great with both JavaScript and TypeScript.

florian-richter replied:
I agree with this

fwilhe commented on DOCUMENTATION.md:
```diff
@@ -1,16 +1,19 @@
-### More Helpful Links
-
-- [All versions of this documentation](https://help.sap.com/viewer/product/SAP_CLOUD_SDK/1.0/en-US)
-- [Product page of the SAP Cloud SDK](https://developers.sap.com/topics/cloud-sdk.html)
```
please check the markdown syntax, the links appear broken (at least on github)

<img width="1172" alt="Screenshot 2023-01-17 at 14 42 38" src="https://user-images.githubusercontent.com/2292245/212914408-4dfd2cf9-6c38-46c5-b750-1172b5e8482c.png">

fwilhe replied:
looks like the br tags are an issue for github, not sure why

florian-richter commented on README.md:
```diff
@@ -145,14 +145,20 @@ Feel free to leave your feedback in form of [GitHub issues](https://github.com/S
 ## Links
-
-- [Documentation portal](https://sap.github.io/cloud-sdk/)
+- [Official support channel](https://github.com/SAP/cloud-sdk-js/issues/new/choose)
+  <br>
+  <br>
+- [Github](https://github.com/SAP/cloud-sdk-js)
+  <br>
+  <br>
+- [SAP Cloud SDK Documentation portal](https://sap.github.io/cloud-sdk)
+- [SAP Cloud SDK Documentation portal - Getting started guide](https://sap.github.io/cloud-sdk/docs/js/getting-started)
+- [SAP Cloud SDK Documentation portal - API documentation](https://sap.github.io/cloud-sdk/api/latest)
+- [SAP Cloud SDK Documentation portal - Error handling](https://sap.github.io/cloud-sdk/docs/js/features/error-handling)
+- [SAP Cloud SDK Documentation portal - Release notes](https://sap.github.io/cloud-sdk/docs/js/release-notes)
+  <br>
+  <br>
```
```suggestion
- [Official support channel](https://github.com/SAP/cloud-sdk-js/issues/new/choose)
- [Github](https://github.com/SAP/cloud-sdk-js)
- [SAP Cloud SDK Documentation portal](https://sap.github.io/cloud-sdk)
  - [SAP Cloud SDK Documentation portal - Getting started guide](https://sap.github.io/cloud-sdk/docs/js/getting-started)
  - [SAP Cloud SDK Documentation portal - API documentation](https://sap.github.io/cloud-sdk/api/latest)
  - [SAP Cloud SDK Documentation portal - Error handling](https://sap.github.io/cloud-sdk/docs/js/features/error-handling)
  - [SAP Cloud SDK Documentation portal - Release notes](https://sap.github.io/cloud-sdk/docs/js/release-notes)
```
[req] Like Florian mentioned, the <br>'s seem to not work so well. I also personally don't like the look. Indentation may be a good alternative.

florian-richter commented on DOCUMENTATION.md:
```diff
@@ -1,16 +1,19 @@
-- [SAP Cloud SDK community page](https://community.sap.com/topics/cloud-sdk)
-- [Community Call for SAP Cloud SDK](https://blogs.sap.com/2019/06/26/sap-cloud-sdk-new-format-of-the-update-call-for-sap-community/)
+## Links
+- [Official support channel](https://github.com/SAP/cloud-sdk-js/issues/new/choose)
+<br>
+<br>
+- [Github](https://github.com/SAP/cloud-sdk-js)
+<br>
+<br>
+- [SAP Cloud SDK Documentation portal](https://sap.github.io/cloud-sdk)
+- [SAP Cloud SDK Documentation portal - Getting started guide](https://sap.github.io/cloud-sdk/docs/js/getting-started)
+- [SAP Cloud SDK Documentation portal - API documentation](https://sap.github.io/cloud-sdk/api/latest)
+- [SAP Cloud SDK Documentation portal - Error handling](https://sap.github.io/cloud-sdk/docs/js/features/error-handling)
+- [SAP Cloud SDK Documentation portal - Release notes](https://sap.github.io/cloud-sdk/docs/js/release-notes)
+<br>
+<br>
+- [Sample repository](https://github.com/SAP-samples/cloud-sdk-js)
```
```suggestion
- [Official support channel](https://github.com/SAP/cloud-sdk-js/issues/new/choose)
- [Github](https://github.com/SAP/cloud-sdk-js)
- [SAP Cloud SDK Documentation portal](https://sap.github.io/cloud-sdk)
  - [SAP Cloud SDK Documentation portal - Getting started guide](https://sap.github.io/cloud-sdk/docs/js/getting-started)
  - [SAP Cloud SDK Documentation portal - API documentation](https://sap.github.io/cloud-sdk/api/latest)
  - [SAP Cloud SDK Documentation portal - Error handling](https://sap.github.io/cloud-sdk/docs/js/features/error-handling)
  - [SAP Cloud SDK Documentation portal - Release notes](https://sap.github.io/cloud-sdk/docs/js/release-notes)
- [Sample repository](https://github.com/SAP-samples/cloud-sdk-js)
```

florian-richter commented on packages/connectivity/README.md:
```diff
@@ -33,22 +33,20 @@ We try to provide fixes, features and answers as soon as possible.
-
-<br>
 
+- [Official support channel](https://github.com/SAP/cloud-sdk-js/issues/new/choose)
+  <br>
+  <br>
```
```suggestion
- [Official support channel](https://github.com/SAP/cloud-sdk-js/issues/new/choose)
```

florian-richter commented on packages/connectivity/README.md:
```diff
@@ -33,22 +33,20 @@ We try to provide fixes, features and answers as soon as possible.
-
-<br>
-
+  <br>
+  <br>
```
```suggestion
```

florian-richter commented on packages/connectivity/README.md:
```diff
@@ -33,22 +33,20 @@ We try to provide fixes, features and answers as soon as possible.
-
-<br>
-
+  <br>
+  <br>
```
```suggestion
```
+ indent the specific documentation links (so all but first) above

florian-richter commented on tsconfig.typedoc.json:
```diff
@@ -76,7 +76,7 @@
       "packages/temporal-de-serializers/src/index.ts"
     ],
-    "name": "SAP Cloud SDK for JavaScript / TypeScript",
+    "name": "SAP Cloud SDK for JavaScript (and TypeScript)",
```
```suggestion
    "name": "SAP Cloud SDK for JavaScript",
```
[pp] Like Florian mentioned, the official name is SAP Cloud SDK for JavaScript. We can just leave out the TS part.

florian-richter reviewed (changes requested):
I would remove all `<br>` tags and use indentation to make the list easier to read. Also, if we change the name, I agree that we should just drop the TypeScript part to match the official name.

FrankEssenberger commented:
I have changed the links and remove the `ErrorHandling` since this was a rather specific documentation topic and added the overview page instead.  I also removed the (and Typescript) which I will also adjust on the landing page.

florian-richter reviewed (approved):
LGTM
